llvm/lib/Transforms/InstCombine/InstCombineShifts.cpp

```diff
@@ -1284,6 +1284,14 @@ Instruction *InstCombinerImpl::visitLShr(BinaryOperator &I) {
     return NewSub;
   }
 
+  // Fold (X + Y) / 2 --> (X & Y & 1) iff (X u<= 1) && (Y u<= 1)
```

Comment:
```suggestion
  // Fold (X + Y) / 2 --> (X & Y) iff (X u<= 1) && (Y u<= 1)
```

https://alive2.llvm.org/ce/z/WF_7mX

Reply:
It looks unrelated. Don't worry :)